File: Stage30/Python/Problem1644.py
import sys

input = sys.stdin.readline

def main():
    N = int(input())
    
    is_prime = [True] * (N + 1)
    if N >= 0: is_prime[0] = False
    if N >= 1: is_prime[1] = False
    
    for i in range(2, int(N**0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, N + 1, i):
                is_prime[j] = False
    
    primes = [i for i in range(2, N + 1) if is_prime[i]]
    
    left, right = 0, 0
    sum_val = 0
    count = 0
    
    while True:
        if sum_val >= N:
            if sum_val == N:
                count += 1
            sum_val -= primes[left]
            left += 1
        else:
            if right == len(primes):
                break
            sum_val += primes[right]
            right += 1
    
    print(count)

if __name__ == "__main__":
    main()

File: Stage30/C/Problem1644.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main(void) {
    int N;
    scanf("%d", &N);

    bool* isPrime = (bool*)malloc(sizeof(bool) * (N + 1));
    
    for (int i = 0; i <= N; i++) isPrime[i] = true;
    if (N >= 0) isPrime[0] = false;
    if (N >= 1) isPrime[1] = false;

    for (int i = 2; i * i <= N; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= N; j += i) {
                isPrime[j] = false;
            }
        }
    }

    int* primes = (int*)malloc(sizeof(int) * (N + 1));
    int size = 0;

    for (int i = 2; i <= N; i++) {
        if (isPrime[i]) {
            primes[size++] = i;
        }
    }

    int left = 0, right = 0;
    int sum = 0;
    int count = 0;

    while (1) {
        if (sum >= N) {
            if (sum == N) count++;
            sum -= primes[left++];
        } else {
            if (right == size) break;
            sum += primes[right++];
        }
    }

    printf("%d\n", count);

    free(isPrime);
    free(primes);
    return 0;
}